safe/gui/tools/print_report_dialog.py

```diff
@@ -0,0 +1,486 @@
+
+            self.composer = self.iface.createNewComposer()
+
+            with open(template_path) as template_file:
+                template_content = template_file.read()
+
+            document = QtXml.QDomDocument()
+            document.setContent(template_content)
+            qgis_composition_context = impact_report.qgis_composition_context
+
+            # load composition object
+            composition = QgsComposition(qgis_composition_context.map_settings)
+            load_status = composition.loadFromTemplate(document)
+
+            if not load_status:
+                # noinspection PyCallByClass,PyTypeChecker
+                QtGui.QMessageBox.warning(
+                    self,
+                    self.tr('InaSAFE'),
+                    self.tr('Error loading template: %s') % template_path)
+
+                return
+
+            self.composer.setComposition(composition)
+            self.composer.composerWindow().show()
+
+    def accept(self):
+        """Method invoked when OK button is clicked."""
+
+        self.save_state()
+        # Get selected template path to use
+        if self.default_template_radio.isChecked():
+            self.selected_template_path = self.template_combo.itemData(
+                self.template_combo.currentIndex())
+        else:
+            self.selected_template_path = self.template_path.text()
+            if not exists(self.selected_template_path):
+                # noinspection PyCallByClass,PyTypeChecker
+                QtGui.QMessageBox.warning(
+                    self,
+                    self.tr('InaSAFE'),
+                    self.tr('Please select a valid template before printing. '
+                            'The template you choose does not exist.'))
+
+        self.dock.show_busy()
+
+        generated_components = [
+            override_component_template(
+                map_report, self.selected_template_path),
+            infographic_report
+        ]
+        error_code, message = self.impact_function.generate_report(
+            generated_components, IFACE=self.iface)
+
+        if error_code == ImpactReport.REPORT_GENERATION_FAILED:
+            self.dock.hide_busy()
+            LOGGER.info(tr(
+                'The impact report could not be generated.'))
+            send_error_message(self, message)
+            LOGGER.info(message.to_text())
+            return ANALYSIS_FAILED_BAD_CODE, message
+
+        sender_name = self.sender().objectName()
+
+        try:
+            if sender_name == 'button_print_pdf':
+                self.create_pdf = True
+                self.print_as_pdf()
+            else:
+                self.create_pdf = False
+                self.open_in_composer()
+            self.dock.hide_busy()
+        except Exception as e:
+            self.dock.hide_busy()
+
+        QtGui.QDialog.accept(self)
+
+    @pyqtSignature('')  # prevents actions being handled twice
+    def on_template_chooser_clicked(self):
```

Comment:
We also move away from these auto-connect slots. Can you use explicit connections and remove the decorator.